src/backend/distributed/transaction/worker_transaction.c

```diff
@@ -329,6 +331,86 @@ SendCommandToMetadataWorkersParams(const char *command,
 }
 
 
+/*
+ * SendCommandToWorkersOptionalInParallel sends the given command to workers in parallel.
+ * It does error if there is a problem while sending the query, but it doesn't error
+ * if there is a problem while executing the query.
+ */
+void
+SendCommandToWorkersOptionalInParallel(TargetWorkerSet targetWorkerSet, const
+									   char *command,
+									   const char *user)
+{
+	ListCell *connectionCell = NULL;
+
+	List *connectionList = OpenConnectionsToWorkersInParallel(targetWorkerSet, user);
+
+	/* finish opening connections */
+	FinishConnectionListEstablishment(connectionList);
+
+	/* send commands in parallel */
+	foreach(connectionCell, connectionList)
+	{
+		MultiConnection *connection = (MultiConnection *) lfirst(connectionCell);
+
+		SendRemoteCommand(connection, command);
+	}
+
+	GetConnectionsResultsOptional(connectionList);
+}
+
+
+/*
+ * OpenConnectionsToWorkersInParallel opens connections to the given target worker set in parallel,
+ * as the given user.
+ */
+static List *
+OpenConnectionsToWorkersInParallel(TargetWorkerSet targetWorkerSet, const char *user)
+{
+	ListCell *workerNodeCell = NULL;
+	List *connectionList = NIL;
+
+	List *workerNodeList = TargetWorkerSetNodeList(targetWorkerSet, ShareLock);
+
+	foreach(workerNodeCell, workerNodeList)
+	{
+		WorkerNode *workerNode = (WorkerNode *) lfirst(workerNodeCell);
+		char *nodeName = workerNode->workerName;
+		int nodePort = workerNode->workerPort;
+		int32 connectionFlags = OUTSIDE_TRANSACTION;
+
+		MultiConnection *connection = StartNodeUserDatabaseConnection(connectionFlags,
+																	  nodeName, nodePort,
+																	  user, NULL);
+		connectionList = lappend(connectionList, connection);
+	}
+	return connectionList;
+}
+
+
+/*
+ * GetConnectionsResultsOptional gets remote command results
+ * for the given connections. It doesn't raise any error.
+ */
+static void
+GetConnectionsResultsOptional(List *connectionList)
+{
+	ListCell *connectionCell = NULL;
+
+	foreach(connectionCell, connectionList)
+	{
+		MultiConnection *connection = (MultiConnection *) lfirst(connectionCell);
+		bool raiseInterrupt = false;
+		PGresult *result = GetRemoteCommandResult(connection, raiseInterrupt);
+		if (result != NULL && IsResponseOK(result))
```

Comment:
we still need to PQclear if !IsReponseOK.

Reply:
@marcocitus, I moved the PQClear outside the if check.